core/vm/common.go

```diff
@@ -28,6 +28,8 @@ import (
 // Global Debug flag indicating Debug VM (full logging)
 var Debug bool
 
+var GenerateStructLogs bool = false
```

Comment:
I agree and I had been thinking about refactoring this (including the `Debug` global). The idea is that the environment should return `Config() vm.Config` which can than be used by the vm to determine these type of things. This will allow for all type of funky and cool stuff like flags, op code callbacks, etc.